tests/unit/components/sl-chart-test.js

```diff
@@ -0,0 +1,363 @@
+import Ember from 'ember';
+import { moduleForComponent, test } from 'ember-qunit';
+
+let testOptions = {
+    chartOptions: {
+        chart: {
+            type: 'bar'
+        },
+
+        xAxis: {
+            categories: [ 'Apples', 'Bananas', 'Oranges' ]
+        },
+
+        yAxis: {
+            title: {
+                text: 'Fruit Eaten'
+            }
+        }
+    }
+};
+
+let testSeries = [{ 
+    name: 'Alice', 
+    data: [ 1, 0, 4 ]
+}];
+
+moduleForComponent( 'sl-chart', 'Unit | Component | sl chart', {
+	unit: true
+});
+
+test( 'Default classNames are present', function( assert ) {
+    this.subject({ 
+        options: testOptions, 
+        series: testSeries 
+    });
+
+    assert.ok(
+        this.$().hasClass( 'panel' ),
+        'Default rendered component has class "panel"'
+    );
+
+    assert.ok(
+        this.$().hasClass( 'panel-default' ),
+        'Default rendered component has class "panel-default"'
+    );
+
+    assert.ok(
+        this.$().hasClass( 'sl-chart' ),
+        'Default rendered component has class "sl-chart"'
+    );
+
+    assert.ok(
+        this.$().hasClass( 'sl-panel' ),
+        'Default rendered component has class "sl-panel"'
+    );
+});
+
+test( 'Loading state adds loading class', function( assert ) {
+    let component = this.subject({
+        options: testOptions, 
+        series: testSeries 
+    });
+
+    assert.strictEqual(
+        this.$().hasClass( 'sl-loading' ),
+        false,
+        'Default rendered component does not have "sl-loading" class'
+    );
+
+    Ember.run( () => {
+        component.set( 'isLoading', true );
+    });
+
+    assert.ok(
+        this.$().hasClass( 'sl-loading' ),
+        'Rendered component gains "sl-loading" class'
+    );
+});
+
+test( 'updateData() is called once', function( assert ) {
```

Comment:
A more inclusive test would be to check that `updateData()` is called when the `series` property's value changes.  It's true that `updateData()` is called during the initial render of the component but this is only because `series` is initially `null` and the observer defined in `updateData()` is setup to watch `series` and then when the series data passed into the component is set on the `series` property this observer fires again (as an observer watching this property would do).  So it's more about that fact that this observer has been setup correctly and responds to changes to `series` *as well as* that this method is called during the initial rendering of the component.

Additionally, there are no tests for the logic contained within this method.  What about the if/else logic inside the for loop?  Should test that neither of the options in the if/else are called when `series` has not length.  Then test each branch of the if/else.  This method does not return a value so it's not as easy as asserting against a call to it - the side-effects of its execution need to be observed and asserted against.